### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by svitalis123**
## [@svitalis123 @Daphineatim @Maxwell011 ] we Accomplished
✔️  We Created a console app that will help you to keep a record of different types of things you own: books, music albums, movies, and games. Created from a UML class diagram. The data was stored in JSON files we also prepared database with tables structure analogical to our program's class structure. Created unit tests for the methods.

Our updated video on description is https://drive.google.com/file/d/186OYhlh1D4pc8nfZnnV_mG3bIbEkdihv/view?usp=sharing

### CODE REVIEW

File: spec/author_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  it 'should have a first name' do
+    expect(@new_author.first_name).to be_a(String)
+  end
+
+  it 'should have a last name' do
+    expect(@new_author.last_name).to be_a(String)
+  end
+end
```
Comment (ShadyShawkat) [inline spec/author_spec.rb]: - [ ] Kindly make sure that all public methods in all classes have unit tests that cover all edge cases, as per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#:~:text=Add%20unit%20tests%20for%20all%20implemented%20methods.) requirement.

File: preserve.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  def load_authors
+    saved_authors = JSON.parse(File.read('storage_files/saved_authors.json'))
+    saved_authors.each do |author|
+      new_author = Author.new(author['first_name'], author['last_name'])
+      @authors << new_author
+    end
+  end
+end
```
Comment (ShadyShawkat) [inline preserve.rb]: - [ ] Please make sure that the app works as intended if one of the JSON files is missing. Currently, if one of the JSON files is missing, the newly created records are saved temporarily in the app until the app is closed and the data is lost when the app is reopened.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
